crates/jiter/src/number_decoder.rs

```diff
@@ -348,18 +360,21 @@ pub(crate) static INT_CHAR_MAP: [bool; 256] = {
     ]
 };
 
-pub struct NumberRange;
+pub enum NumberRange {
+    Int(Range<usize>),
+    Float(Range<usize>),
```

Comment:
I wonder whether this would be better served by a `NumberRangeType` enum plus a `Range<usize>` member, instead of an enum.

Reply:
changed.